src/routeGeneration/routeGenerator.ts

```diff
@@ -38,57 +35,68 @@ export class RouteGenerator {
     });
   }
 
-  public GenerateExpressRoutes() {
-    return this.GenerateRoutes(expressTemplate, path => path.replace(/{/g, ':').replace(/}/g, ''));
+  public GenerateCustomRoutes(template: string, pathTransformer: (path: string) => string) {
+    let file: string;
+    fs.readFile(path.join(template), (err, data) => {
+      if (err) {
+        throw err;
+      }
+      file = data.toString();
+      return this.GenerateRoutes(file, pathTransformer);
+    });
   }
 
-  public GenerateHapiRoutes() {
-    return this.GenerateRoutes(hapiTemplate, path => path);
-  }
+  private buildContent(middlewareTemplate: string, pathTransformer: (path: string) => string) {
+    handlebars.registerHelper('json', function (context: any) {
+      return JSON.stringify(context);
+    });
 
-  public GenerateKoaRoutes() {
-    return this.GenerateRoutes(koaTemplate, path => path.replace(/{/g, ':').replace(/}/g, ''));
-  }
+    const routesTemplate = handlebars.compile(`
+      /* tslint:disable */
+      {{#if canImportByAlias}}
+        import { ValidateParam } from 'tsoa';
+        import { Controller } from 'tsoa';
+      {{else}}
+        import { ValidateParam } from '../../../src/routeGeneration/templateHelpers';
+        import { Controller } from '../../../src/interfaces/controller';
+      {{/if}}
+      {{#if iocModule}}
+      import { iocContainer } from '{{iocModule}}';
+      {{/if}}
+      {{#each controllers}}
+      import { {{name}} } from '{{modulePath}}';
+      {{/each}}
+
+      const models: any = {
+        {{#each models}}
+        "{{name}}": {
+          {{#each properties}}
+            "{{@key}}": {{{json this}}},
+          {{/each}}
+        },
+        {{/each}}
+      };`
+      .concat(middlewareTemplate), { noEscape: true });
 
-  private buildContent(middlewareTemplate: string, pathTransformer: (path: string) => string) {
+    const authenticationModule = this.options.authenticationModule ? this.getRelativeImportPath(this.options.authenticationModule) : undefined;
+    const iocModule = this.options.iocModule ? this.getRelativeImportPath(this.options.iocModule) : undefined;
+
+    // If we're working locally then tsoa won't exist as an importable module.
+    // So, we reference its helpers and base controller via path instead of alias.
+    // Note: require.resolve('tsoa') works under normal circumstances, but when using npm link
+    // to develop locally, it breaks. Keep this in mind.
     let canImportByAlias: boolean;
     try {
-      require('tsoa');
+      require.resolve('tsoa');
```

Comment:
This feels problematic if you're trying to test with a project that is linked locally to tsoa. I tested, and if using link - require always throws an error.

If this is only there to support the tests running - could we perhaps use NODE_ENV instead to set the environment to test - and change the imports that way instead?